### ISSUE TITLE
Covers from GDrive aren't visible from time to time

### ISSUE THREAD

**Issue by OzzieIsaacs** [janeczku/calibre-web#2236]
Now I have the problem with the covers that happened to me before. I don't know if I should close this issue and open another one or if I can continue here, just in case I expose the problem. If you think you have to close it go ahead, I will open a specific one for it.

I can not see the covers of the books, it is an intermittent error, because from time to time when I enter the library I can see them, but if I browse for a while they end up failing. I can't identify the error, since I don't see anything in the logs.

![image](https://user-images.githubusercontent.com/17300743/147235298-5dd43247-9cf5-4fd4-a2f3-246763e91daf.png)

**Logs**
```
[2021-12-23 11:22:31,081]  INFO {cps.server:272} webserver stop (restart=True)
[2021-12-23 11:22:32,249]  INFO {cps.server:260} Performing restart of Calibre-Web
[2021-12-23 11:22:35,708] DEBUG {cps.services:35} Cannot import simpleldap, logging in with ldap will not work: No module named 'flask_simpleldap'
[2021-12-23 11:22:41,596]  INFO {cps:133} *** "requests" version does not fit the requirements. Should: <2.25.0, Found: 2.25.0, please consider installing required version ***
[2021-12-23 11:22:41,598]  INFO {cps:141} Starting Calibre Web...
[2021-12-23 11:22:42,032]  INFO {cps.server:204} Starting Gevent server on [::]:443
[2021-12-23 11:22:53,110] DEBUG {cps.updater:326} Nightly version: bdedec90dd8df032cbc666a26618f17bcde3b6ea, 2021-12-23T11:13:08+01:00
[2021-12-23 11:23:01,129] DEBUG {cps.config_sql:337} _ConfigSQL updating storage
[2021-12-23 11:23:18,330]  INFO {oauth2client.client:2074} Successfully retrieved access token
[2021-12-23 11:23:23,588] DEBUG {cps.config_sql:337} _ConfigSQL updating storage
[2021-12-23 11:23:42,198] DEBUG {cps.updater:326} Nightly version: bdedec90dd8df032cbc666a26618f17bcde3b6ea, 2021-12-23T11:13:08+01:00
[2021-12-23 11:32:49,756] DEBUG {cps.updater:326} Nightly version: bdedec90dd8df032cbc666a26618f17bcde3b6ea, 2021-12-23T11:13:08+01:00`
```

_Originally posted by @jagalile in https://github.com/janeczku/calibre-web/issues/2233#issuecomment-1000245721_

**Comment by OzzieIsaacs** [comment]
Please check the developer console of your browser:
![image](https://user-images.githubusercontent.com/6438895/147236189-48671b5a-ee0e-4f20-ae37-84c1f552f3c9.png)
I guess in this case it will look similar to my one (except I bet for an 404 error). Please post it here

**Comment by jagalile** [comment]
I get this

![image](https://user-images.githubusercontent.com/17300743/147237405-97b8d8e8-0969-4682-8a3c-b319c27654f5.png)

If I click on the link _drive.google.com/uc?......_ I download the book cover, there may be problems if the file name contains spaces or _( )_

![image](https://user-images.githubusercontent.com/17300743/147237805-59e970cd-ecdf-457f-8021-ed8dc8a07d1b.png)

**Comment by OzzieIsaacs** [comment]
On GDrive every file/folder has an id, something like: 1HHpBpYu_H1p7xi1s-B-OcOY39Oiyx9hP. Calibre-Web starts a request to gdrive and transforms the path into this Id. Then it returns a redirect request to your browser, so it tries to load the file from google drive. Errorcode 403 means "forbidden". Normally the first request should lead to a authentification on gdrive, so all covers are allowed to download. Could you please open the gdrive folder in the same browser you are viewing calibre-web. I guess it's working afterwards.
If there is no reverse proxy in between eating some request, this looks like something never worked...

**Comment by jagalile** [comment]
Ok, I have found the problem and it has nothing to do with the configuration or the repository.

By telling me that the Errorcode 403 means "forbidden" I realized that my default browser was blocking third party cookies and that usually gives problems with google services.

It's all fixed now, sorry for wasting your time with something that is not directly related to this repository. At least I hope that if it happens to someone else I can learn from this, thank you very much for the help.
